### PR TITLE
fix(host search): fix search parameters

### PR DESCRIPTION
**Pull request by miteto**
- replace search_h and search_s with searchH and searchS
- add empty Search parameter
- make additional checks for historySearch (fix warnings)
- make additional checks in header (fix warnings)

<h1> Fix search parameters </h1>

<h2> fix typos of parameters and warnings for non-existing properties </h2>

Fixes 3568
<h2> Type of change </h2>

- [X] Patch fixing an issue (non-breaking change)

<h2> Target serie </h2>

- [ ] 2.8.x
- [X] 18.10.x
- [X] 19.04.x (master)

### CODE REVIEW

**Comment by sc979** [comment]
@vhr ,
Please take your time when you read a PR.
As I told dimitar, the coding style stipulate that we use camelCase : So we shouldn't use $_GET['Search'].
In the htmlHeader.php file, dimitar modified the line 128 an not the line 125 !

File: www/include/core/header/htmlHeader.php [hunk 1]
```diff
@@ -125,8 +125,12 @@
     if (isset($_GET["search"])) {
```
Comment (sc979) [inline www/include/core/header/htmlHeader.php]: this won't work as you replaced $_GET['search'] with $_GET['Search']

File: www/include/core/header/htmlHeader.php [hunk 2]
```diff
@@ -125,8 +125,12 @@
...
         $searchStr .= "search_host=" . htmlentities($_GET["search"], ENT_QUOTES, "UTF-8");
     }
-    if (isset($centreon->historySearch[$url]) && !isset($_GET["search"])) {
-        $searchStr .= "search_host=" . $centreon->historySearch[$url];
+    if (isset($centreon->historySearch[$url]) && !isset($_GET["Search"])) {
```
Comment (sc979) [inline www/include/core/header/htmlHeader.php]: use camelCase for "Search"

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 3]
```diff
@@ -86,13 +86,13 @@
     $centreon->historySearch[$url]["hostStatus"] = $hostStatus;
 } elseif (isset($_GET['Search'])) {
     $centreon->historySearch[$url] = array();
-    $template = $_GET['template'];
+    $template = $_GET['template'] ?? '';
```
Comment (sc979) [inline www/include/configuration/configObject/service/listServiceByHost.php]: you already now that $_GET['Search'] is set

File: www/include/configuration/configObject/host/listHost.ihtml [hunk 4]
```diff
@@ -67,7 +67,7 @@
 			<a href="{$elemArr[elem].RowMenu_link}"><img src="{$elemArr[elem].RowMenu_icone}" class="ico-18  margin_right" />{$elemArr[elem].RowMenu_name}</a>
 		</td>
 		<td class="ListColCenter">
-            <a href='./main.php?p=602&search_h={$elemArr[elem].RowMenu_name}&search_s='>
+            <a href='./main.php?p=602&Search=&searchH={$elemArr[elem].RowMenu_name}&searchS='>
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.ihtml]: use camelCase for "Search"

File: www/include/configuration/configObject/service/listServiceByHost.php (hunk 3 above)
Comment (sc979) [reply www/include/configuration/configObject/service/listServiceByHost.php]: $template is initialized on line 60, using null

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 5]
```diff
@@ -84,15 +84,15 @@
...
     isset($_POST["statusHostFilter"]) ? $hostStatus = 1 : $hostStatus = 0;
     $centreon->historySearch[$url]["hostStatus"] = $hostStatus;
-} elseif (isset($_GET['Search'])) {
+} elseif (isset($_GET['search'])) {
     $centreon->historySearch[$url] = array();
-    $template = $_GET['template'];
+    $template = $_GET['template'] ?? '';
     $centreon->historySearch[$url]['template'] = $template;
-    $status = $_GET["status"];
+    $status = $_GET["status"] ?? '';
```
Comment (sc979) [inline www/include/configuration/configObject/service/listServiceByHost.php]: status should be set to -1 to display both status.

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 6]
```diff
@@ -84,15 +84,15 @@
...
     $centreon->historySearch[$url]["status"] = $status;
     $searchH = $_GET["searchH"];
     $centreon->historySearch[$url]["searchH"] = $searchH;
-    $searchS = $_GET["searchS"];
+    $searchS = $_GET["searchS"] ?? '';
```
Comment (sc979) [inline www/include/configuration/configObject/service/listServiceByHost.php]: searchS was initialized to null on line 66

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 7]
```diff
@@ -84,15 +84,15 @@
     $centreon->historySearch[$url]["searchS"] = $searchS;
     isset($_POST["statusHostFilter"]) ? $hostStatus = 1 : $hostStatus = 0;
     $centreon->historySearch[$url]["hostStatus"] = $hostStatus;
-} elseif (isset($_GET['Search'])) {
+} elseif (isset($_GET['search'])) {
```
Comment (sc979) [inline www/include/configuration/configObject/service/listServiceByHost.php]: as you can see $_GET['search'] and $_POST['Search'] do the same thing. why do you keep duplicated code, when you could modify line 75 and remove the elseif ?

Comment (sc979) [reply www/include/configuration/configObject/service/listServiceByHost.php]: you didn't mofidied the line 75 : $_POST['Search'] -> $_POST['search']
(how can it work on your env ??)

File: www/include/core/header/htmlHeader.php [hunk 8]
```diff
@@ -126,7 +126,11 @@
         $searchStr .= "search_host=" . htmlentities($_GET["search"], ENT_QUOTES, "UTF-8");
     }
     if (isset($centreon->historySearch[$url]) && !isset($_GET["search"])) {
-        $searchStr .= "search_host=" . $centreon->historySearch[$url];
+        $searchStr .= "search_host=" . (
+                is_array($centreon->historySearch[$url]) && isset($centreon->historySearch[$url]['searchH']) ?
+                    $centreon->historySearch[$url]['searchH'] :
+                    $centreon->historySearch[$url]
```
Comment (sc979) [inline www/include/core/header/htmlHeader.php]: historySearch is an array of arrays. So (in the else case of your ternary), you're affecting an array to "search_host=" (line 129), and it won't work

File: www/include/configuration/configObject/service/listServiceByHost.php (hunk 6 above)
Comment (miteto) [reply www/include/configuration/configObject/service/listServiceByHost.php]: the old code was $searchS = $_GET["searchS"] which produced PHP warning. Fixed now.

File: www/include/core/header/htmlHeader.php (hunk 8 above)
Comment (miteto) [reply www/include/core/header/htmlHeader.php]: the old code was "$searchStr .= "search_host=" . $centreon->historySearch[$url];" which produced PHP warning.

Comment (sc979) [reply www/include/core/header/htmlHeader.php]: if : "isset($centreon->historySearch[$url]['searchH']" isn't set -> you send the whole array $centreon->historySearch[$url] ?? this won't work

File: www/include/configuration/configObject/host/listHost.ihtml [hunk 9]
```diff
@@ -67,7 +67,7 @@
 			<a href="{$elemArr[elem].RowMenu_link}"><img src="{$elemArr[elem].RowMenu_icone}" class="ico-18  margin_right" />{$elemArr[elem].RowMenu_name}</a>
 		</td>
 		<td class="ListColCenter">
-            <a href='./main.php?p=602&search_h={$elemArr[elem].RowMenu_name}&search_s='>
+            <a href='./main.php?p=602&search=&searchH={$elemArr[elem].RowMenu_name}&searchS='>
```
Comment (sc979) [inline www/include/configuration/configObject/host/listHost.ihtml]: please remove "&search" as it's not a filter, nor used in this page

File: www/include/configuration/configObject/service/listServiceByHost.php [hunk 10]
```diff
@@ -72,29 +72,18 @@
...
-    $centreon->historySearch[$url]["searchH"] = $searchH;
-    $searchS = $_POST["searchS"];
-    $centreon->historySearch[$url]["searchS"] = $searchS;
-    isset($_POST["statusHostFilter"]) ? $hostStatus = 1 : $hostStatus = 0;
-    $centreon->historySearch[$url]["hostStatus"] = $hostStatus;
-} elseif (isset($_GET['Search'])) {
-    $centreon->historySearch[$url] = array();
-    $template = $_GET['template'];
+if (isset($_POST['search']) || isset($_GET['search'])) {
+    $data = isset($_POST['search']) ? $_POST : $_GET;
```
Comment (sc979) [inline www/include/configuration/configObject/service/listServiceByHost.php]: I checked again this page and the XHR request : 
When you do a search, a POST request is sent with the data
The GET is only used to initialize the page.
So no need to keep the isset($_GET['search']), as it won't happen
Don't need to add a new variable $data neither. Use the $_POST[" xx "]  directly

**Comment by sc979** [comment]
@miteto ,
again, please check my PR : https://github.com/centreon/centreon/pull/7348

File: www/include/configuration/configObject/host/listHost.ihtml (hunk 9 above)
Comment (sc979) [reply www/include/configuration/configObject/host/listHost.ihtml]: for this task you only need to modify the line 70 as this :
`<a href='./main.php?p=602&searchH={$elemArr[elem].RowMenu_name}'>`

**Pull request closed without merging**